The release process has a dependency on shx

### 🤔 What's the problem you've observed?

```
$ polyglot-release 20.0.0
sh: 1: shx: not found
npm ERR! code 127
npm ERR! path /home/mpkorstanje/Projects/cucumber/messages/javascript
npm ERR! command failed
npm ERR! command sh -c -- shx echo "// This file is automatically generated using npm scripts" > src/version.ts && echo "export const version = '${npm_package_version}'" >> src/version.ts

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mpkorstanje/.npm/_logs/2022-11-14T14_19_39_663Z-debug-0.log
```

Because we invoke some shell scripts around the build:

https://github.com/cucumber/messages/blob/f4c626728f4715af6c228e81f875dc0db0ad8510/javascript/package.json#L35


### ✨ Do you have a proposal for making it better?

Either remove the (need for) shell scripts or add a check for `shx` to `polyglot-release`.

https://github.com/cucumber/polyglot-release/blob/fc50e6fdf3ab2f8af5f140064887667eab9785e7/polyglot-release#L125-L127

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
+### Fixed
+- [JavaScript] Update static version file at release time ([#459](https://github.com/cucumber/messages/pull/459))
 
 ## [33.0.3] - 2026-06-20
 ### Added

--- a/javascript/package.json
+++ b/javascript/package.json
@@ -31,7 +31,8 @@
     "postbuild": "node -e \"require('.')\"",
     "prepare": "make generate",
     "test": "mocha",
-    "prepublishOnly": "npm run build && npm run copy-schemas"
+    "prepublishOnly": "npm run build && npm run copy-schemas",
+    "version": "printf \"export const version = '%s' as string\\n\" \"$npm_package_version\" > src/version.ts"
   },
   "dependencies": {},
   "devDependencies": {

--- a/javascript/src/version.ts
+++ b/javascript/src/version.ts
@@ -1,3 +1,1 @@
-import { createRequire } from 'node:module'
-
-export const version: string = createRequire(import.meta.url)('../package.json').version
+export const version = '33.0.3' as string

